test/CalculatorW3GTest.cpp

```diff
@@ -0,0 +1,144 @@
+////////////////////////////////////////////////////////////////////////////
+/**
+ *  @file   CalculatorGtest.cpp
+ *  @date   Wed May 26 2021
+ *  @brief  Calculator Unit Tests
+ */
+////////////////////////////////////////////////////////////////////////////
+
+#include<gtest/gtest.h>
+#include<gmock/gmock.h>
+#include "ICalculatorW3.h"
+#include "SimpleExpressionCalculatorW3Factory.h"
+#include "AddCalculatorW3.h"
+#include "SubtractCalculatorW3.h"
+#include "MultiplyCalculatorW3.h"
+#include "DivideCalculatorW3.h"
+
+using namespace testing;
+using namespace calculator;
+
+/**
+ * @brief Fixture for Simple expression calulcators
+ * 
+ */
+class SimpleExpressionCalculatorW3Test : public ::testing::Test 
+{
+    protected:
+        AddCalculatorW3 addCalculator = AddCalculatorW3(10,10);
+        SubtractCalculatorW3 subtractCalculator = SubtractCalculatorW3(10, 10);
+        MultiplyCalculatorW3 multiplyCalculator = MultiplyCalculatorW3(10, 10);
+        DivideCalculatorW3 divideCalculator = DivideCalculatorW3(10, 10);
+};
+
+/**
+ * @brief fixtures for simple expression calculator factory
+ * 
+ */
+class SimpleExpressionCalculatorW3FactoryTest : public ::testing::Test
+{
+    protected:
+        SimpleExpressionCalculatorW3Factory simpleExpressionFactory;
+};
+
+/**
+ * @brief Test to make sure division by zero is handled correctly
+ * 
+ */
+TEST(CalculatorW3Tests, WhenDivideByZeroDivideCalculatorIsConstructed_ThenErrorPrintedToStdError)
+{
+    testing::internal::CaptureStderr();
+    DivideCalculatorW3 divideCalculator = DivideCalculatorW3(10, 0);
+    std::string output = testing::internal::GetCapturedStderr();
+    ASSERT_NE(output, std::string(""));
+}
```

Comment:
We need unit test for :

1. divide a negative number like 10/ -1
2. Divide 2 negative number
3. Divide 0 by 0
4. Divide 0 by negative number